git-sync/main.go

```diff
@@ -168,12 +184,26 @@ func syncRepo(repo, dest, branch, rev string, depth int) error {
 
 	if *flChmod != 0 {
 		// set file permissions
-		_, err = runCommand("chmod", "", []string{"-R", string(*flChmod), dest})
+		_, err = runCommand("chmod", "", []string{"-R", string(*flChmod), pullTarget})
 		if err != nil {
 			return err
 		}
 	}
 
+	if atomic && strings.Compare(pullTarget, path.Join(dest, subdir)) != 0 {
+		args := []string{"-snf", pullTarget, "data-link"}
+		_, err := runCommand("ln", dest, args)
+		if err != nil {
+			return fmt.Errorf("error creating symbolic link: %v", err)
+		}
+
```

Comment:
I was following pseudocode from thockin that seemed to imply ln was not atomic. After a little bit of research, it seems that "ln -f" replaces existing symlinks, which is not atomic, where "mv" is guaranteed atomic. However, if there is no existing symlink at the path, the single "ln" appears to be atomic (from my primitive test). We will want to do ln + mv for refresh pulls though.